packages/node/src/models.ts

```diff
@@ -0,0 +1,117 @@
+import { legacy } from "@counterfactual/cf.js";
+import {
+  Address,
+  AppInstanceInfo,
+  AssetType
+} from "@counterfactual/common-types";
+
+import { zeroBalance } from "./utils";
+
+import Nonce = legacy.utils.Nonce;
+import FreeBalance = legacy.utils.FreeBalance;
+
+/**
+ * The schema of a channel is below.
+
+/**
+ * The fully expanded schema for a channel:
+ * multisigAddress: {
+ *  multisigAddress: Address,
+ *  multisigOwners: Address[],
+ *  rootNonce: Nonce,
+ *  appInstances: Map<AppInstanceID,
+ *    appInstance: {
+ *      id: string,
+ *      appId: Address,
+ *      abiEncodings: {
+ *        stateEncoding: string,
+ *        actionEncoding: string
+ *      },
+ *      appState: any,
+ *      localNonceCount: number,
+ *      uninstallationNonce: {
+ *        isSet: boolean,
+ *        salt: string,
+ *        nonceValue: number
+ *      },
+ *      timeout: BigNumber,
+ *      asset: {
+ *        assetType: AssetType,
+ *        token?: Address
+ *      },
+ *      deposits: Map<Address, BigNumber>
+ *    }
+ *  },
+ *  proposedAppInstances: same schema as appInstances,
+ *  freeBalances: Map<AssetType,
+ *    freeBalance: {
+ *      alice: Address,
+ *      aliceBalance: BigNumber,
+ *      bob: Address,
+ *      bobBalance: BigNumber,
+ *      localNonceCount: number,
+ *      timeout: number,
+ *      dependencyNonce: {
+ *        isSet: boolean,
+ *        salt: string,
+ *        nonceValue: number
+ *      }
+ *    }
+ *  }
+ * }
+ */
+
+/**
+ * This class is only a type implementation of a channel schema for the
+ * purposes of updating and retrieving a channel's state from the store.
+ *
+ * An instance is by itself stateless and effectively reflects the state of an
+ * according channel in the store.
+ */
+export class Channel {
+  constructor(
+    readonly multisigAddress: Address,
+    readonly multisigOwners: Address[],
+    readonly rootNonce: Nonce = new Nonce(true, 0, 0),
```

Comment:
Yeah definitely, might as well add it here though I suppose, even though this isn't actually being used yet.

Reply:
FWIW my comment on a previous PR stands - a root nonce with 0 timeout doesn't make sense

Reply:
Agreed. What do you think is a good default? @IIIIllllIIIIllllIIIIllllIIIIllllIIIIll @snario

Reply:
Actually none of those parameters seem to be concerned with timeout.

They're the `uniqueId` and `nonceValue` used for interacting with the `NonceRegistry`.

Having said that, `nonceValue` being 0 makes sense. `uniqueId` being 0 does not.

I suppose the `uniqueId` used to represent this nonce in the `NonceRegistry` could be the same as the `multisigAddress`, since that's the unique qualifier for the channel the root nonce is operating in? @IIIIllllIIIIllllIIIIllllIIIIllllIIIIll @snario